include/picongpu/particles/atomicPhysics/ionizationPotentialDepression/StewartPyattIPD.hpp

```diff
@@ -255,20 +272,57 @@ namespace picongpu::particles::atomicPhysics::ionizationPotentialDepression
             // UNIT_ENERGY/eV
             constexpr float_X eV = sim.pic.get_eV();
 
+            float_X const chargeState_X = static_cast<float_X>(chargeState);
+
             // UNIT_MASS * UNIT_LENGTH^3 / UNIT_TIME^2 * 1/(eV * UNIT_ENERGY/eV * UNIT_LENGTH)
             // = unitless, not weighted
             float_X const K
                 = (pmacc::math::isApproxZero(
                       superCellConstantInput.temperatureTimesk_Boltzman * superCellConstantInput.debyeLength))
                       ? 0._X
-                      : constFactor * (chargeState + 1)
+                      : constFactor * (chargeState_X + 1._X)
                             / (superCellConstantInput.temperatureTimesk_Boltzman * eV
                                * superCellConstantInput.debyeLength);
 
             // eV, not weighted
-            return superCellConstantInput.temperatureTimesk_Boltzman
-                   * (math::pow((3 * (superCellConstantInput.zStar + 1) * K + 1), 2._X / 3._X) - 1._X)
-                   / (2._X * (superCellConstantInput.zStar + 1._X));
+            float_X const stewartPyattIPD
+                = superCellConstantInput.temperatureTimesk_Boltzman
+                  * (math::pow((3._X * (superCellConstantInput.zStar + 1._X) * K + 1._X), 2._X / 3._X) - 1._X)
```

Comment:
where is this coming from? Probably particle from the link below. Please document better.